src/include/OSL/oslquery.h

```diff
@@ -75,10 +75,10 @@ class OSLQUERYPUBLIC OSLQuery {
         std::vector<ustring> fields;     ///< Names of this struct's fields
         ustring structname;              ///< Name of the struct
         std::vector<Parameter> metadata; ///< Meta-data about the param
-        Parameter ()
-            : isoutput(false), validdefault(false), varlenarray(false),
-              isstruct(false), isclosure(false), data(NULL)
-        { }
+
+        Parameter () {}
```

Comment:
Is there any need to declare these? Don't you just want the "default" behavior?

Reply:
For the copy and move constructors, it's important that the `data` pointer refers to ITS OWN default field, not a straight pointer copy from the source. So default ctrs won't work.

Reply:
ok with this?

Reply:
Ah I missed that detail.

LGTM!